apps/hash-graph/lib/graph/src/api/rest/entity.rs

```diff
@@ -81,9 +88,14 @@ impl RoutedResource for EntityResource {
             "/entities",
             Router::new()
                 .route("/", post(create_entity::<S, A>).put(update_entity::<S, A>))
-                .route(
-                    "/:entity_id/public",
-                    post(make_entity_public::<A>).delete(make_entity_private::<A>),
+                .nest(
+                    "/:entity_id",
+                    Router::new()
+                        .route(
+                            "/public",
+                            post(make_entity_public::<A>).delete(make_entity_private::<A>),
+                        )
+                        .route("/update", get(can_update_entity::<A>)),
```

Comment:
I wonder if there's a better route we can use for this that makes clear it's a permission check... if we want to have a 'can a user view' check, the equivalent to this would be `GET /entities/:entity_id/view`, which I think is ambiguous.

I don't have any great ideas but something we can think about as we add more routes. Maybe `GET /entities/:entity_id/permission/:permission`?

Reply:
Or `GET /entities/:entity_id/authorization/:permission`?